packages/chain-adapters/src/evm/avalanche/AvalancheChainAdapter.ts

```diff
@@ -0,0 +1,206 @@
+import {
+  ASSET_REFERENCE,
+  AssetId,
+  avalancheAssetId,
+  avalancheChainId,
+  fromAssetId,
+  toAssetId
+} from '@shapeshiftoss/caip'
+import { bip32ToAddressNList, ETHSignTx } from '@shapeshiftoss/hdwallet-core'
+import { BIP44Params, KnownChainIds } from '@shapeshiftoss/types'
+import * as unchained from '@shapeshiftoss/unchained-client'
+import BigNumber from 'bignumber.js'
+import { numberToHex } from 'web3-utils'
+
+import { ErrorHandler } from '../../error/ErrorHandler'
+import { BuildSendTxInput, FeeDataEstimate, GasFeeDataEstimate, GetFeeDataInput } from '../../types'
+import { toPath } from '../../utils'
+import { bn, bnOrZero } from '../../utils/bignumber'
+import { ChainAdapterArgs, EVMBaseAdapter } from '../EVMBaseAdapter'
+import { Fees } from '../types'
+import { getErc20Data } from '../utils'
+
+const SUPPORTED_CHAIN_IDS = [avalancheChainId]
+const DEFAULT_CHAIN_ID = avalancheChainId
+
+export class ChainAdapter extends EVMBaseAdapter<KnownChainIds.AvalancheMainnet> {
+  public static readonly defaultBIP44Params: BIP44Params = {
+    purpose: 44,
+    coinType: Number(ASSET_REFERENCE.AvalancheC),
+    accountNumber: 0
+  }
+
+  constructor(args: ChainAdapterArgs) {
+    super({ chainId: DEFAULT_CHAIN_ID, supportedChainIds: SUPPORTED_CHAIN_IDS, ...args })
+
+    this.assetId = toAssetId({
```

Comment:
use hardcoded chain id over creating?

Reply:
this would allow support for testnet ultimately since we are using `this.chainId` to create the `assetId`

Reply:
I guess we have hardcoded mainnet everywhere else though, so I suppose we could for now if we wanted to